SonarQube.TeamBuild.PreProcessor/TeamBuildPreProcessor.cs

```diff
@@ -19,21 +21,27 @@ public class TeamBuildPreProcessor
         public const string FxCopCSharpRuleset = "SonarQubeFxCop-cs.ruleset";
         public const string FxCopVBNetRuleset = "SonarQubeFxCop-vbnet.ruleset";
 
+        private readonly ILogger logger;
         private readonly ISonarQubeServerFactory serverFactory;
         private readonly ITargetsInstaller targetInstaller;
+        private readonly IAnalyzerProvider analyzerProvider;
 
         #region Constructor(s)
 
-        public TeamBuildPreProcessor()
-            : this(new SonarQubeServerFactory(), new TargetsInstaller())
+        public TeamBuildPreProcessor(ILogger logger)
+            : this(logger, new SonarQubeServerFactory(), new TargetsInstaller(), new RoslynAnalyzerProvider(logger))
```

Comment:
We want the logger to be passed to the analyser provider on construction so we need to pass it to the pre-processor on construction too.